noxfile.py

```diff
@@ -0,0 +1,61 @@
+from pathlib import Path
+
+import nox
+from nox import Session, session
+
+python_versions = ['3.5', '3.6', '3.7', '3.8', '3.9', '3.10']
+nox.options.sessions = ['tests', 'docs', 'style']
+nox.options.reuse_existing_virtualenvs = True
+
+
+@session(python=python_versions)
+def tests(session: Session) -> None:
+    """Run the tests."""
+    args = session.posargs or ['--buffer']
+
+    session.install('--upgrade', 'pip', 'setuptools', 'wheel', 'coverage')
+    session.install('-e', '.')
+
+    try:
+        session.run(
+            'coverage', 'run', '--parallel', '--module', 'unittest', *args
```

Comment:
On the one hand I see that --parallel is handy, on the other does that mean that you could run into outdated coverage files accidentally getting combined by the combine step?  I don't use nox myself so I have no idea how best to handle this; perhaps something like (inventing API here, but you get the idea)
```
def coverage(...):
    python_versions = ...  # something from the command line?
    for version in python_versions: tests(version, coverage_file=<known_path>)
    session.run('coverage', 'combine', <known_paths>)
```
?

Reply:
I've never really had an issue with it, and you could always remove all .coverage files first.  Coverage combine removes all of the files that it combines together, so assume it runs then there would only be a single .coverage file leftover.  And coverage combine always ignores that file and overwrites it the next time.  You could potentially end up with a extra .coverage.* file if the nox session was stopped early or something and then it would be added in the coverage combine the next time it was run.

Also, Nox doesn't actually run in parallel, it runs them all sequentially.  I'm just used to using parallel when I setup stuff in CI.  So we could change this to just use --append and let each subsequent python version just append to the single .coverage file.  Perhaps we would need to first delete the coverage file though?

One option would be to make a separate session that simply removes all coverage files first, then triggers the test session (similar to how the test session triggers the coverage one in the current implementation).

Another option would be to just make it all explicit: remove the parallel flag and define the coverage file for each run using the COVERAGE_FILE environment variable.  This is similar to what you do in GitHub actions currently after re-reviewing it.

Reply:
Let's go for COVERAGE_FILE then.

Reply:
Added